src/Services/Notification.vala

```diff
@@ -101,6 +104,23 @@ public class Notifications.Notification : Object {
             app_icon = app_info.get_icon ().to_string ();
         }
 
+        // GLib.Notification.set_icon ()
+        image_path = lookup_string (hints, "image-path");
+        if (image_path != null) {
+            // Ensure we're not being sent symbolic badge icons
+            image_path = image_path.replace ("-symbolic", "");
```

Comment:
Am I being paranoid that this isn't explicitly only replaced when it's at the end? If some icon was `poorly-symbolic-named` this would behave unexpectedly.

Reply:
no because you can also have `-rtl` variants, so it has to be able to remove `-symbolic` from the "middle" of the string to keep that working as well

Reply:
Ah that's right. Yay string based matching specs!